File: problem2/solution.py
class Solution:
    def jump(self, nums: list[int]) -> int:
        jump_count = 0
        goalIndex = len(nums) - 1
        while goalIndex != 0:
            for i, num in enumerate(nums):
                if (i + num >= goalIndex):
                    jump_count += 1
                    goalIndex = i
                    break
        return jump_count



def main():
    solution = Solution()
    jump_count = solution.jump([2,3,1,1,4])
    print("jumps: ",jump_count)

if __name__ == "__main__":
    main()

File: problem2/Solution.java
class Solution{
    public int jump(int[] nums) {
        int goalIndex = nums.length - 1;
        int jumpCount = 0;

        while (goalIndex != 0){
            for (int i = 0; i<nums.length; i++){
                if (i + nums[i] >= goalIndex){
                    jumpCount++;
                    goalIndex = i;
                    break;
                }
            }
        }
        return jumpCount;
    }

    public static void main(String[] args){
        Solution solution = new Solution();
        int[] nums = {2,3,1,1,4};
        int jumps = solution.jump(nums);
        System.out.println("jumps: " + jumps);
    }
}